src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
```

Comment:
PullProofAI

The guard throws an `UnauthorizedException` when the feedback secret is invalid. This behavior might leak sensitive information about the application's internal workings.

#### Recommendation
Avoid exposing internal details through exceptions. Instead, provide a generic error message to the client without revealing specifics.

```diff
-     throw new UnauthorizedException('Invalid feedback secret');
+     throw new UnauthorizedException('Unauthorized');
```


<!-- This is an auto-generated comment by PullProofAI -->